File: src/main/java/com/timepath/hl2/io/bsp/lump/Edge.java
package com.timepath.hl2.io.bsp.lump;

import com.timepath.io.struct.StructField;
import org.jetbrains.annotations.NotNull;

import java.util.logging.Logger;

public class Edge {

    private static final Logger LOG = Logger.getLogger(Edge.class.getName());
    /**
     * Vertex indices
     */
    @NotNull
    @StructField
    public short[] v = new short[2];

    public Edge() {
    }
}

File: src/main/kotlin/com/timepath/hl2/io/bsp/lump/Edge.kt
package com.timepath.hl2.io.bsp.lump

import com.timepath.hl2.io.bsp.Lump
import com.timepath.hl2.io.bsp.LumpHandler
import com.timepath.io.OrderedInputStream
import com.timepath.io.struct.Struct
import com.timepath.io.struct.StructField

private val MAX_MAP_EDGES = 256000

public class Edge(
        /** Vertex indices */
        @StructField(0) val v: ShortArray = ShortArray(2)
) {
    class Handler : LumpHandler<List<Edge>> {
        override fun invoke(l: Lump, ois: OrderedInputStream): List<Edge> {
            return (0..(l.length / Struct.sizeof(Edge())) - 1).map {
                ois.readStruct<Edge>(Edge())
            }
        }
    }
}
